app.py

```diff
@@ -0,0 +1,98 @@
+from flask import Flask, redirect, render_template, url_for, request, session
+from flask_mysqldb import MySQL
+import MySQLdb.cursors
+import MySQLdb.cursors, re, hashlib
+from datetime import date
+app = Flask(__name__)
+
+app.secret_key = 'arremangala arrepujala sí, arremangala arrepujala no'
+
+app.config['MYSQL_HOST'] = 'localhost'
+app.config['MYSQL_USER'] = 'root'
+app.config['MYSQL_PASSWORD'] = ''
+app.config['MYSQL_DB'] = 'mindhive'
+
+# Initialize MySQL
+mysql = MySQL(app)
+
+@app.route("/")
+def route():
+    return redirect(url_for('login'))
+
+@app.route("/login", methods=['GET', 'POST'])
+def login():
+    
+    msg = ''
+    # Check if username and password fields exist in the form
+    if request.method == 'POST' and 'username' in request.form and 'password' in request.form:
+        # Variables creation
+        username = request.form['username']
+        password = request.form['password']
+
+        #Password Hashing
+        hash = password + app.secret_key
+        hash = hashlib.sha1(hash.encode())
+        password = hash.hexdigest()
+        # Attempt to log-in
+        cursor = mysql.connection.cursor(MySQLdb.cursors.DictCursor)
+        cursor.execute('SELECT * FROM user WHERE email = %s AND password_hash = %s', (username, password,))
+        
+        account = cursor.fetchone()
+        # If account exists in db
+        if account:
+            # Creates session data
+            session['loggedin'] = True
+            session['uid'] = account['uid']
+            session['email'] = account['email']
+            return 'Inicio de sesión exitoso'
+        else:
+            # Log-in failure
+            msg = 'Email o contraseña incorrectas'
+    return render_template('/auth/login.html', msg=msg)
+
+
+@app.route('/register', methods=['GET', 'POST'])
+def register():
+
+    msg = ''
+    # Check if required fields exist in the form
+    if request.method == 'POST' and 'email' in request.form and 'nombre' in request.form and 'apellido' in request.form and "password" in request.form and "confirmarPassword" in request.form:
```

Comment:
I would rewrite this snippet instead of the current for to avoid repetition:
```python
if (request.method == 'POST')
 and (all(field in request.form for field in ['email', 'nombre', 'apellido', 'password', 'confirmarPassword'])):
# Code goes here
else:
# Code if any of the fields are not present in the form
```
This snippet would allow us to check if all the fields in the list (`['email', 'nombre', 'apellido', 'password', 'confirmarPassword']`) are present as keys in the dictionary `request.form`.

What do you think?

Reply:
looks neat!